### PR TITLE
optionally ignore extra fields when deserializing

### PR DESCRIPTION
**Pull request by ipa-jba**
closed one PR, opened another.

We often ran into troubles that we used the message converter to connect two systems with different "capabilities". If one non-ros System send a message which contained additional fields compared to "our" definition of the ros message, the message converter raises a ValueError. With the additional option `strict_mode` for converting json or dictionaries to ros messages extra fields can just be ignored.

Default behavior stays as before. The new feature needs to be explicitly enabled

### CODE REVIEW

File: test/test_message_converter.py [hunk 1]
```diff
@@ -286,6 +286,22 @@ def test_dictionary_with_empty(self):
         expected_message = serialize_deserialize(expected_message)
         self.assertEqual(message, expected_message)
 
+    def test_dictionary_with_empty_additional_args_strict_mode(self):
+        from std_msgs.msg import Empty
+        dictionary = {"additional_args": "should raise value error"}
+        with self.assertRaises(ValueError) as context:
+            message_converter.convert_dictionary_to_ros_message('std_msgs/Empty', dictionary)
+        print(context.exception.message.find("has no field named") >= 0)
```
Comment (mintar) [inline test/test_message_converter.py]: The print statement should be removed.

Comment (ipa-jba) [reply test/test_message_converter.py]: leftover from getting the test to run --> removed and changed the commit

**Comment by ipa-jba** [comment]
The pipeline fails are in indigo and lunar, how to deal with those?

**Comment by mintar** [comment]
> The pipeline fails are in indigo and lunar, how to deal with those?

Just rebase onto current master, I've removed indigo and lunar just now. The tests fail because those distros are EOL and have been removed from rosdep.

**Comment by mintar** [comment]
Thanks for the PR! Merged.

**Pull request merged**